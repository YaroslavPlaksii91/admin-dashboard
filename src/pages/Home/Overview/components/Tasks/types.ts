export type TasksItemType = {
  title: string;
  priority: string;
  isFinished: boolean;
};

export type TasksProps = {
  data: TasksItemType[];
};
