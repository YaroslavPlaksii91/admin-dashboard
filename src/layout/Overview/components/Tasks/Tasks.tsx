import { FC } from 'react';
import { List, ListItem, Typography, Box, SvgIcon } from '@mui/material';

import { Link } from '@components/Link/Link';
import { Label } from '@components/Label/Label';
import { LABEL_TYPES } from '@components/Label/constants';
import { CheckboxComponent } from '@components/Checkbox/Checkbox';

import { TasksProps } from './types';

export const Tasks: FC<TasksProps> = ({ data }) => {
  return (
    <Box
      sx={{
        width: '50%',
        padding: '32px 0 8px',
        border: '1px solid',
        borderColor: 'borderColor',
        borderRadius: '8px',
        backgroundColor: 'mainWhiteColor',
      }}
    >
      <Box
        sx={{
          display: 'flex',
          justifyContent: 'space-between',
          mb: '8px',
          padding: '0 32px',
        }}
      >
        <Typography variant="h3">Tasks</Typography>
        <Link url="#">View all</Link>
      </Box>

      <Typography
        sx={{
          fontSize: '12px',
          lineHeight: 1.3,
          letterSpacing: 0.1,
          mb: '30px',
          padding: '0 32px',
          color: 'secondaryTextColor',
        }}
      >
        Today
      </Typography>

      <Box
        sx={{
          display: 'flex',
          justifyContent: 'space-between',
          alignItems: 'center',
          padding: '0 32px',
          mb: '18px',
        }}
      >
        <Typography
          sx={{
            fontSize: '14px',
            fontWeight: 600,
            lineHeight: 1.43,
            letterSpacing: 0.2,
            color: 'additionalInfoColor',
          }}
        >
          Create new task
        </Typography>
        <SvgIcon titleAccess="Add new task" sx={{ cursor: 'pointer' }}>
          <use href="/src/assets/icons/sprite.svg#icon-create"></use>
        </SvgIcon>
      </Box>

      <List>
        {data.map(item => (
          <ListItem
            key={item.title}
            sx={{
              justifyContent: 'space-between',
              padding: '16px 32px',
              borderTop: '1px solid',
              borderTopColor: 'borderColor',
            }}
          >
            <Box sx={{ display: 'flex', gap: 4 }}>
              <CheckboxComponent />

              <Typography variant="subtitle2" component="p">
                {item.title}
              </Typography>
            </Box>

            <Label
              text={item.priority}
              type={LABEL_TYPES.SQUARE}
              color={item.priority}
            />
          </ListItem>
        ))}
      </List>
    </Box>
  );
};
