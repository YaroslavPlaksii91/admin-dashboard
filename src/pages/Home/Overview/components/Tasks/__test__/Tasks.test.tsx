import { cleanup, render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import { Tasks } from '../Tasks';

jest.mock('react-router-dom', () => {
  const originalModule = jest.requireActual('react-router-dom');
  return {
    ...originalModule,
    useNavigate: () => jest.fn(),
  };
});

describe('Tasks component', () => {
  afterEach(cleanup);

  const mockData = [
    { title: 'Task 1', priority: 'high', isFinished: true },
    { title: 'Task 2', priority: 'medium', isFinished: false },
    { title: 'Task 3', priority: 'low', isFinished: false },
  ];

  it('should be rendered', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Tasks data={mockData} />
      </BrowserRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the task list', () => {
    render(
      <BrowserRouter>
        <Tasks data={mockData} />
      </BrowserRouter>,
    );

    mockData.forEach(item => {
      expect(screen.getByText(item.title)).toBeInTheDocument();
    });
  });

  it('displays "Create new task" button', () => {
    render(
      <BrowserRouter>
        <Tasks data={mockData} />
      </BrowserRouter>,
    );

    expect(screen.getByText('Create new task')).toBeInTheDocument();
  });

  it('displays "View all" link', () => {
    render(
      <BrowserRouter>
        <Tasks data={mockData} />
      </BrowserRouter>,
    );

    expect(screen.getByText('View all')).toBeInTheDocument();
  });
});
