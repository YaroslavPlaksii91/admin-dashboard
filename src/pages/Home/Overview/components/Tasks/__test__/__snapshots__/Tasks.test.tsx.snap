// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tasks component should be rendered 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1wjthba"
  >
    <div
      class="MuiBox-root css-tgxzo7"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
      >
        Tasks
      </h3>
      <a
        href="/"
      >
        View all
      </a>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1i9dev7-MuiTypography-root"
    >
      Today
    </p>
    <div
      class="MuiBox-root css-70phyn"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-18jlpju-MuiTypography-root"
      >
        Create new task
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          focusable="false"
          role="img"
          viewBox="0 0 24 24"
        >
          <use
            href="/sprite.svg#icon-create"
          />
          <title>
            Add new task
          </title>
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <ul
      class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
    >
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1an86nx-MuiListItem-root"
      >
        <div
          class="MuiBox-root css-13t3iqc"
        >
          <label>
            <span
              class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-le7zt4-MuiButtonBase-root-MuiCheckbox-root"
              data-testid="checkbox"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <div
              class="MuiBox-root css-yhnhoj"
            >
              <svg
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1xhbeow-MuiSvgIcon-root"
                focusable="false"
                role="img"
                viewBox="0 0 24 24"
              >
                <use
                  href="/src/assets/icons/sprite.svg#icon-inactiveCheckbox"
                />
                <title>
                  Task not completed
                </title>
              </svg>
            </div>
          </label>
          <p
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Task 1
          </p>
        </div>
        <span
          class="undefined"
        >
          high
        </span>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1an86nx-MuiListItem-root"
      >
        <div
          class="MuiBox-root css-13t3iqc"
        >
          <label>
            <span
              class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-le7zt4-MuiButtonBase-root-MuiCheckbox-root"
              data-testid="checkbox"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <div
              class="MuiBox-root css-yhnhoj"
            >
              <svg
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1xhbeow-MuiSvgIcon-root"
                focusable="false"
                role="img"
                viewBox="0 0 24 24"
              >
                <use
                  href="/src/assets/icons/sprite.svg#icon-inactiveCheckbox"
                />
                <title>
                  Task not completed
                </title>
              </svg>
            </div>
          </label>
          <p
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Task 2
          </p>
        </div>
        <span
          class="undefined"
        >
          medium
        </span>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1an86nx-MuiListItem-root"
      >
        <div
          class="MuiBox-root css-13t3iqc"
        >
          <label>
            <span
              class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-le7zt4-MuiButtonBase-root-MuiCheckbox-root"
              data-testid="checkbox"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <div
              class="MuiBox-root css-yhnhoj"
            >
              <svg
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1xhbeow-MuiSvgIcon-root"
                focusable="false"
                role="img"
                viewBox="0 0 24 24"
              >
                <use
                  href="/src/assets/icons/sprite.svg#icon-inactiveCheckbox"
                />
                <title>
                  Task not completed
                </title>
              </svg>
            </div>
          </label>
          <p
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Task 3
          </p>
        </div>
        <span
          class="undefined"
        >
          low
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
